pyflakes/test/test_other.py

```diff
@@ -13,6 +13,130 @@ class Test(TestCase):
     def test_duplicateArgs(self):
         self.flakes('def fu(bar, bar): pass', m.DuplicateArgument)
 
+    def test_duplicate_keys(self):
+        self.flakes("{'yes': 1, 'yes': 2}", m.DuplicateDictionaryKey)
+
+    @skipIf(version_info >= (3,),
+            "bytes and strings with same 'value' are not equal in python3")
+    def test_duplicate_keys_bytes_vs_unicode(self):
```

Comment:
It would be helpful to do a similar py2 check with `{1: b'a', 1: u'a'}` .

It also wouldnt hurt adding py3 checks, confirming that it treats b'a' and u'a' as non-equal, even though that is one of the defining characteristics of py3. It would be more for regression testing to prevent someone royally screwing up in the future if they modified your algorithm.